Library/Homebrew/cask/artifact/exec_script.rb

```diff
@@ -0,0 +1,67 @@
+# typed: true # rubocop:todo Sorbet/StrictSigil
+# frozen_string_literal: true
+
+require "cask/artifact/binary"
+
+module Cask
+  module Artifact
+    # Artifact corresponding to the `exec_script` stanza.
+    class ExecScript < Binary
+      DEFAULT_ARGS = ["$@"].freeze
+
+      def self.from_args(cask, source, **args)
+        new(cask, source, **args)
+      end
+
+      sig {
+        params(
+          cask:       Cask,
+          source:     T.any(String, Pathname),
+          # File name for installed script. The default is the basename of `source`
+          target:     T.nilable(String),
+          # File name for wrapper script. The default is `#{target}.wrapper.sh`
+          wrapper:    T.nilable(String),
+          # Arguments added to exec script for source command
+          args:       T.nilable(T::Array[T.any(String, Pathname)]),
+          # Set to `false` to disable adding double quotes around each argument in `args`
+          quote_args: T::Boolean,
+          # The shell for shebang. The default is `/bin/bash`
+          shell:      String,
+          # A location to send standard error. Nothing by default. Can be set to `/dev/null` or a file
+          stderr:     T.nilable(T.any(String, Pathname)),
```

Comment:
```suggestion
```

Will remove this as only used in 1 cask (`p4v`) and can just use `args` instead if needed.